docs/framework-decorator-client.md

```diff
@@ -0,0 +1,224 @@
+    base_url="https://api.example.com",
+    headers={"Authorization": "Bearer <token>"},
+    timeout=10.0,
+)
+client = Client(config=config)
+
+# Convenience shortcut
+client = Client(base_url="https://api.example.com")
+```
+
+The `Config` mirrors the familiar options from generated clients (base URL, headers, timeout, authentication, redirects, etc.) so you can keep the same mental model for HTTP behavior.
+
+Use the same client instance for sync or async handlers:
+
+```python
+client = Client(base_url="https://api.example.com")
+
+# sync
+@client.get("/ping")
+def ping(result: str) -> str:
+    return result
+
+# async
+@client.get("/version")
+async def version(result: str) -> str:
+    return result
+
+assert ping() == "pong"
+version_string = await version()
+```
+
+## GET example
+
+```python
+from typing import Optional
+from pydantic import BaseModel
+from clientele import Client
+
+client = Client(base_url="https://api.example.com")
+
+class User(BaseModel):
+    id: int
+    name: str
+    email: Optional[str]
+
+@client.get("/users/{user_id}")
+def get_user(user_id: int, include_details: bool = True, result: User, response=None) -> User:
+    assert response.status_code == 200
+    return result
+
+user = get_user(42)
+```
+
+How it works:
+
+- Path parameters inside `{}` are filled from the function arguments (e.g. `user_id`).
+- Any remaining keyword arguments become query parameters unless you provide `query={...}`.
+- The return type (`-> User`) drives response parsing; there is no need for a separate `response_model` argument.
+- If your function accepts a `result` parameter, the parsed payload is injected. If your function accepts a `response` parameter, the underlying `httpx.Response` is injected.
+
+### Path + query together
+
+Borrowing from the example OpenAPI spec at `example_openapi_specs/best.json` (the `/request-data/{path_parameter}` path), you can blend path substitution with explicit query parameters:
+
+```python
+from pydantic import BaseModel
+from clientele import Client
+
+client = Client(base_url="https://api.example.com")
+
+class RequestDataResponse(BaseModel):
+    path_parameter: str
+    your_query: str
+
+@client.get("/request-data/{path_parameter}")
+def fetch_request_data(path_parameter: str, *, result: RequestDataResponse) -> RequestDataResponse:
+    return result
+
+response = fetch_request_data("some-id", query={"your_query": "hello"})
+assert response.path_parameter == "some-id"
+assert response.your_query == "hello"
+```
+
+## Class-based APIs (Pattern A)
+
+The functional decorator style above still works as-is. If you prefer class-based APIs with per-instance configuration, use ``Routes`` to declare endpoints at import time and attach a ``Client`` to each instance:
+
+```python
+from clientele import Client, Config, Routes
+from pydantic import BaseModel
+
+routes = Routes()
+
+
+class User(BaseModel):
+    id: int
+    name: str
+
+
+class UsersAPI:
+    def __init__(self, base_url: str):
```

Comment:
I don't like asking the user to set an `__init__` - can we use an abc class somehow to instead enforce declared values on the model or a declare config?